### PR TITLE
No noDipoleInSW

### PR DESCRIPTION
**Pull request by ykempf**
Warn when noDipoleInSW is 1 in Magnetosphere with dipole type 4 as it turns out not to be a good idea.

When removing the dipolar part in the inflow slab of cells there's a curl of BGB that causes spurious Bx growing in the inflow. At coarse resolution it (again) leads to very visible bending of the supposedly purely southward IMF. Less dramatic in production runs like FIC.

The setup below is the upstream of a magnetosphere with dipole type 4 without refinement.

~dev
~this branch

### CODE REVIEW

**Review by alhom — APPROVED** [review]

File: projects/Magnetosphere/Magnetosphere.cpp [hunk 1]
```diff
@@ -109,20 +109,19 @@ namespace projects {
...
 
-      /* Enforce no dipole in solar wind with dipole type 4 */
-      if ((this->dipoleType == 4) && (!this->noDipoleInSW)) {
+      /* Remind the user of what they might want to do, current wisdom is that type 4 needs the dipole in the inflow cells too. */
+      if ((this->dipoleType == 4) && (!this->dipoleInSW)) {
          if(myRank == MASTER_RANK) {
-            std::cerr<<"Note: Initializing Magnetosphere with dipole type 4, enforcing no dipole in solar wind!"<<std::endl;
+            std::cerr<<"Note: Initializing Magnetosphere with dipole type 4, you should probably have the dipole in the solar wind and therefore set dipoleInSW = 1 (default)."<<std::endl;
          }
-         this->noDipoleInSW = true;
```
Comment (ykempf) [inline projects/Magnetosphere/Magnetosphere.cpp:125]: Well, do we want to force it on actually then?

Comment (alhom) [reply projects/Magnetosphere/Magnetosphere.cpp:125]: No forcing, but error msg if not used (as suggested)? Users do, of course, read the outputs...

File: projects/Magnetosphere/Magnetosphere.cpp [hunk 2]
```diff
@@ -117,11 +117,12 @@ namespace projects {
...
       RP::get("Magnetosphere.dipoleType", this->dipoleType);
 
-      /* Actually, rather do not enforce no dipole in solar wind with dipole type 4, but remind the user of what they might want to do. */
+      /* Enforce "dipole" (incl. correction terms) in solar wind with dipole type 4. */
       if ((this->dipoleType == 4) && (this->noDipoleInSW)) {
          if(myRank == MASTER_RANK) {
-            std::cerr<<"Note: Initializing Magnetosphere with dipole type 4, you might want to actually have the dipole in the solar wind and therefore set noDipoleInSW = 0."<<std::endl;
+            std::cerr<<"Note: Initializing Magnetosphere with dipole type 4, you want to have the dipole + vector potential correction terms in the solar wind and therefore set noDipoleInSW = 0."<<std::endl;
          }
+         this->noDipoleInSW = true;
```
Comment (markusbattarbee) [inline projects/Magnetosphere/Magnetosphere.cpp:125]: false

**Review by markusbattarbee — APPROVED** [review]

**Comment by ykempf** [comment]
That's a *jees* I think:
image
image

File: projects/Magnetosphere/Magnetosphere.cpp [hunk 3]
```diff
@@ -117,12 +117,12 @@ namespace projects {
 
       RP::get("Magnetosphere.dipoleType", this->dipoleType);
 
-      /* Enforce no dipole in solar wind with dipole type 4 */
-      if ((this->dipoleType == 4) && (!this->noDipoleInSW)) {
+      /* Enforce "dipole" (incl. correction terms) in solar wind with dipole type 4. */
+      if ((this->dipoleType == 4) && (this->noDipoleInSW)) {
          if(myRank == MASTER_RANK) {
-            std::cerr<<"Note: Initializing Magnetosphere with dipole type 4, enforcing no dipole in solar wind!"<<std::endl;
+            std::cerr<<"Note: Initializing Magnetosphere with dipole type 4, you want to have the dipole + vector potential correction terms in the solar wind and therefore set noDipoleInSW = 0."<<std::endl;
```
Comment (ykempf) [inline projects/Magnetosphere/Magnetosphere.cpp:123]: This message changed but is not gone @ursg?

**Comment by ykempf** [comment]
up

**Pull request merged**